src/main/kotlin/org/rust/cargo/project/CargoProjectDescription.kt

```diff
@@ -27,8 +27,8 @@ class CargoProjectDescription private constructor(
     ) {
         val isModule: Boolean get() = source == null
         val libTarget: Target? get() = targets.find { it.isLib }
-
         val contentRoot: VirtualFile? get() = VirtualFileManager.getInstance().findFileByUrl(contentRootUrl)
+        val isExternal: Boolean get() = targets.asSequence().filter { it.crateRoot != null }.any { it.isLib }
```

Comment:
>The idea here is to separate the source code we control (it's in our workspace) from the source code we don't (external dependencies).

Yes, this is exactly what I want. I think that this implementation of `isExternal` won't work always, and it is quite possible that `isModule` is broken as well. 

There's no need to make it super correct right now, it can always be fixed later. 

The problem with `external == any target is a library` happes when, for example, you are implementing the library. 

If you have

```
src/
  main.rs
  lib.rs
```

and `main.rs` panics